core/src/test/scala/unit/kafka/server/AddPartitionsToTxnManagerTest.scala

```diff
@@ -201,7 +201,7 @@ class AddPartitionsToTxnManagerTest {
     assertEquals(expectedDisconnectedErrors, transaction1Errors)
     assertEquals(expectedDisconnectedErrors, transaction2Errors)
 
-    val expectedTopLevelErrors = topicPartitions.map(_ -> Errors.INVALID_RECORD).toMap
+    val expectedTopLevelErrors = topicPartitions.map(_ -> Errors.INVALID_TXN_STATE).toMap
```

Comment:
Is it useful to have a test case in which we abort the transaction explicitly after seeing this error and then start a new transaction?

Reply:
It could be. It may be a bit tricky to mock fully but I can take a stab at it.

Reply:
I've added the test